cargo-cyclonedx/src/generator.rs

```diff
@@ -62,9 +63,11 @@ use validator::validate_email;
 type PackageMap = BTreeMap<PackageId, Package>;
 type ResolveMap = BTreeMap<PackageId, Node>;
 
-pub struct SbomGenerator {}
+pub struct SbomGenerator<'g> {
+    config: &'g SbomConfig,
```

Comment:
I would prefer to avoid adding a lifetime here. We can just make the SbomGenerator own the struct, that will make it much simpler to work with. An extra `clone()` for  the config will not make any noticeable difference in performance.

Reply:
Thanks, agreed. I was just about to add the very same comment just without a good solution :)

Reply:
Actually the generator can own the config for the time running, and then give it back. So no need for a reference or a clone.